v2/test/Makefile

```diff
@@ -22,14 +22,16 @@ sample-test: tmp/context.tar.gz
 		--gcs_root $(GCS_ROOT)/data \
 		--gcr_root $(GCR_ROOT)
 
-.PHONY: tmp/context.tar.gz
-tmp/context.tar.gz:
-# Archive source folder
-# ref: https://stackoverflow.com/a/12010656
+.PHONY: context
+context:
+	mkdir -p tmp
 # Benefit of using `git archive` over `tar` is that, `--exclude-vcs-ignore` flag
 # we want to use is not supported in MacOS (bsdtar). This feature is only
 # available in gnutar.
-	mkdir -p tmp
-# cd $(REPO_ROOT) && git archive --format=tar $$(git stash create) | gzip >v2/test/tmp/context.tar.gz
-# TODO(Bobgy): figure out how to allow archiving dirty region.
-	cd $(REPO_ROOT) && git archive --format=tar HEAD | gzip >v2/test/tmp/context.tar.gz
+#
+# The following line is adapted based on an one-liner that can archive source
+# folder taking .gitignore into account. This command works no matter your
+# working dir is dirty or not. It always packages current content into the tarball.
+# Credits: https://stackoverflow.com/a/23486703/8745218
+# License: https://creativecommons.org/licenses/by-sa/3.0/
```

Comment:
this is not a correct form of attribution, I'll reimplement this line of code to avoid the license problem